pkg/client/gen.go

```diff
@@ -80,13 +74,28 @@ func (*SonobuoyClient) GenerateManifest(cfg *GenConfig) ([]byte, error) {
 		}
 	}
 
-	plugins := []*manifest.Manifest{}
-	if includes(cfg.Config.PluginSelections, e2ePluginName) {
-		plugins = append(plugins, e2eManifest(cfg))
+	// Support legacy logic for the time being.
+	if len(cfg.DynamicPlugins) == 0 && len(cfg.StaticPlugins) == 0 &&
+		len(cfg.Config.PluginSelections) > 0 {
```

Comment:
Double check this logic and add a test case; if we want to support backwards compat then we need to support the empty array -> run no plugins case. Rare, but that is the intent and we need to make sure it works as expected.

Reply:
Should be `PluginSelections!=nil`